featuretools/entityset/entityset.py

```diff
@@ -1151,10 +1151,27 @@ def add_last_time_indexes(self, updated_dataframes=None):
             self.dataframe_dict[df.ww.name] = df
 
         self.reset_data_description()
+
     # ###########################################################################
-    # #  Other ###############################################
+    # #  Pickling ###############################################
     # ###########################################################################
+    def __getstate__(self):
+        return {
+            **self.__dict__,
+            'schemas': {df_name: df.ww.schema for df_name, df in self.dataframe_dict.items()}
+        }
+
+    def __setstate__(self, state):
+        schemas = state.pop('schemas')
+        for df_name, df in state.get('dataframe_dict', {}).items():
+            if schemas[df_name] is not None:
+                df.ww.init(schema=schemas[df_name])
+        for k, v in state.items():
+            object.__setattr__(self, k, v)
```

Comment:
I don't think it matters one way or another since `self.__dict__[k] = v` uses `__setattr__` anyways. It can be simplified to just `self.__dict__ = state` though so I've updated to use that.

Reply:
Yeah I think that'll be simpler.  It might also make sense to do `self.__dict__.update(state)` just in case there's already something in `__dict__` for some reason.  There also appears to be a bit of precedent for using `update` in the python docs: https://docs.python.org/3/library/pickle.html#handling-stateful-objects.

Reply:
> `self.__dict__[k] = v` uses `__setattr__` anyways

I didn't realize this so I just poked around in the console a bit to confirm.  However, it appears that setting a key on `__dict__` skips over the whole descriptor protocol and method resolution and doesn't actually hit `__setattr__`:
```python
In [1]: class Foo:
   ...:     def set_attr(self, k, v):
   ...:         self.__dict__[k] = v
   ...:     def __setattr__(self, k, v):
   ...:         print("__setattr__ called!")
   ...:         super().__setattr__(k, v)
   ...:

In [2]: f = Foo()

In [3]: f.set_attr("a", 1)

In [4]: vars(f)
Out[4]: {'a': 1}

In [5]: f.a = 2
__setattr__ called!

In [6]: vars(f)
Out[6]: {'a': 2}
```
I thought I'd just mention it here for posterity. :)